triggers/service/authentication.md

```diff
@@ -0,0 +1,95 @@
+# Design for Trigger Service Authentication/Authorization
+
+## Goals
+
+- Be compatible with an OAuth2 authorization code grant
+  https://tools.ietf.org/html/rfc6749#section-4.1
+- Do not require OAuth2 or any other specific
```

Comment:
curious how compatible with the authorization code grant you will be, if you're not exchanging the granted code with specifically an OAuth2 tenant for an access token?

Reply:
Not quite sure what you are getting at. The auth middleware is exchanging the granted code with the authorization server for the token.

Reply:
ok! yea I was a bit overly focused on the semantics of 'compatible' here. this makes sense.